Sources/SwiftLexicalLookup/ScopeImplementations.swift

```diff
@@ -0,0 +1,190 @@
+        LookupName.getNames(from: codeBlockItem.item, accessibleAfter: codeBlockItem.endPosition)
+      }
+    case .memberBlock:
+      return statements.flatMap { codeBlockItem in
+        LookupName.getNames(from: codeBlockItem.item)
+      }
+    }
+  }
+
+  public func lookup(
+    for name: String?,
+    at syntax: SyntaxProtocol,
+    with configDict: LookupConfigDictionary
+  ) -> [LookupResult] {
+    let nameIntroductionStrategy = configDict[FileScopeNameIntroductionStrategy.self] ?? .memberBlockUpToLastDecl
+
+    let names = introducedNames(using: nameIntroductionStrategy)
+      .filter { introducedName in
+        introducedName.isAccessible(at: syntax) && (name == nil || introducedName.refersTo(name!))
+      }
+
+    return [.fromFileScope(self, withNames: names, nameIntroductionStrategy: nameIntroductionStrategy)]
+  }
+}
+
+@_spi(Experimental) extension CodeBlockSyntax: ScopeSyntax {
+  public var introducedNames: [LookupName] {
+    statements.flatMap { codeBlockItem in
+      LookupName.getNames(from: codeBlockItem.item, accessibleAfter: codeBlockItem.endPosition)
+    }
+  }
+}
+
+@_spi(Experimental) extension ForStmtSyntax: ScopeSyntax {
+  public var introducedNames: [LookupName] {
+    LookupName.getNames(from: pattern)
+  }
+}
+
+@_spi(Experimental) extension ClosureExprSyntax: ScopeSyntax {
+  public var introducedNames: [LookupName] {
+    let captureNames =
+      signature?.capture?.children(viewMode: .sourceAccurate).flatMap { child in
+        if let captureList = child.as(ClosureCaptureListSyntax.self) {
+          captureList.children(viewMode: .sourceAccurate).flatMap { capture in
+            LookupName.getNames(from: capture)
+          }
+        } else {
+          LookupName.getNames(from: child)
+        }
+      } ?? []
+
+    let parameterNames =
+      signature?.parameterClause?.children(viewMode: .sourceAccurate).flatMap { parameter in
+        if let parameterList = parameter.as(ClosureParameterListSyntax.self) {
+          parameterList.children(viewMode: .sourceAccurate).flatMap { parameter in
+            LookupName.getNames(from: parameter)
+          }
+        } else {
+          LookupName.getNames(from: parameter)
+        }
+      } ?? []
+
+    return captureNames + parameterNames
+  }
+}
+
+@_spi(Experimental) extension WhileStmtSyntax: ScopeSyntax {
+  public var introducedNames: [LookupName] {
+    conditions.flatMap { element in
+      LookupName.getNames(from: element.condition)
+    }
+  }
+}
+
+@_spi(Experimental) extension IfExprSyntax: ScopeSyntax {
+  public var parentScope: ScopeSyntax? {
+    getParent(for: self.parent, previousIfElse: self.elseKeyword == nil)
+  }
+
+  /// Finds the parent scope, omitting parent `if` statements if part of their `else if` clause.
+  private func getParent(for syntax: Syntax?, previousIfElse: Bool) -> ScopeSyntax? {
+    guard let syntax else { return nil }
+
+    if let lookedUpScope = syntax.scope, lookedUpScope.id != self.id {
```

Comment:
It's worth having a code example here to show a closure with both a capture list and named parameters so it's obvious what names are being introduced